lms/djangoapps/class_dashboard/dashboard_data.py

```diff
@@ -35,14 +56,16 @@ def get_problem_grade_distribution(course_id):
       'total_student_count' where the key is problem 'module_id' and the value is number of students
         attempting the problem
     """
+    non_student_list = get_non_student_list(course_id)
 
     # Aggregate query on studentmodule table for grade data for all problems in course
-    db_query = models.StudentModule.objects.filter(
+    db_query = models.StudentModule.objects.select_related('student').filter(
```

Comment:
I don't think you need to add select_related('student') to this query since StudentModule has student_id column.

Try with:
.exclude(student_id__in=non_student_list)